utils/mcp_client.py

```
@@ -0,0 +1,531 @@
+    host = parsed.hostname
+    try:
+        if _is_private_ip(host):
+            raise McpClientError(f"MCP HTTP URL host is private or local: {host}")
+        return
+    except ValueError:
+        pass
+
+    try:
+        addresses = socket.getaddrinfo(host, None)
+    except socket.gaierror as e:
+        raise McpClientError(f"Could not resolve MCP HTTP host '{host}': {e}") from e
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)